src/App.tsx

```diff
@@ -1,16 +1,163 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  getTodos,
+  addTodo,
+  removeTodo,
+  patchTodo,
+} from './api/todos';
+
+import { Header } from './components/Header/Header';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { Error } from './components/Error/Error';
+import { ErrorContext } from './components/Error/ErrorContext';
+import { Footer } from './components/Footer/Footer';
+
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const {
+    Load,
+    Add,
+    Delete,
+    Update,
+  } = Errors;
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [title, setTitle] = useState('');
+
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const {
+    setHasError,
+    setCurrentError,
+    isAdding,
+    setIsAdding,
+    setSelectedTodoIds,
+  } = useContext(ErrorContext);
+
+  const loadUserTodos = useCallback(async () => {
+    if (user) {
+      try {
+        setHasError(false);
+
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
```

Comment:
```suggestion
        setHasError(false);
        setIsAdding(true);
        setTodos(await getTodos(user.id));
        setIsAdding(false);
```

Reply:
https://user-images.githubusercontent.com/90685701/207912029-06a20392-b471-45d8-9828-26ca31e6f61e.mov

Reply:
that's not exactly the same as in the demo but it's the best I could find for you

this is something related to an empty list of todos and the first addition to that list
check out your code (including css), maybe that hint will help

Reply:
Thanks a million!) <3